shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -603,6 +583,17 @@ - (FlutterPlatformViewController*)platformViewController {
 
 #pragma mark - Private methods
 
+- (FlutterViewController*)viewControllerForId:(uint64_t)viewId {
+  // TODO(dkwingsmt): The engine only supports single-view, therefore it
+  // only processes the default ID. After the engine supports multiple views,
+  // this method should be able to return the view for any IDs.
+  NSAssert(viewId == kFlutterDefaultViewId, @"Unexpected view ID %llu", viewId);
```

Comment:
IIUC, this should be removed when we officially support multi-window - and we're checking here whether the viewId is that of the root window (window 0).

If all of that's true, we might say that "Multiple view controllers aren't currently supported. viewControllerForId got an unexpected viewId: %llu"

Reply:
Well, there is no way for the user to send a request with a different ID anyway. The framework doesn't support so. So this is really just an internal error.